src/shared/contexts/pinneditems.tsx

```diff
@@ -0,0 +1,151 @@
+import React, {
+  createContext,
+  FC,
+  useEffect,
+  useState,
+  memo,
+  useCallback,
+  ReactElement,
+} from 'react'
+import {CLOUD} from '../constants'
+
+let getPinned
+let putPinned
+let deletePinned
+let postPinned
+
+if (CLOUD) {
+  const pinnedItemFunctions = require('src/client/pinnedItemRoutes')
+  getPinned = pinnedItemFunctions.getPinned
+  putPinned = pinnedItemFunctions.putPinned
+  postPinned = pinnedItemFunctions.postPinned
+  deletePinned = pinnedItemFunctions.deletePinned
+}
```

Comment:
this looks fishy to me. We're saying to require this routes file which only exists for cloud and assign the imported functions to these variables. But the rest of this file assumes those variables are defined...which isn't the case for OSS right?

Reply:
Changed this around so that its handled anywhere where we make this import. Wont get imported and/or used unless its both CLOUD and the flag for pinned items is true. That way, there is no way this can be accessed or used without it being in CLOUD

Reply:
LMK your thoughts

Reply:
if this file is only imported when CLOUD is set, can't we just import the functions from `pinnedItemRoutes` normally in this file?

Reply:
yes, you are correct - i should change this now